### ISSUE TITLE
vue/attribute-hyphenation doesn't work on <slot> element

### ISSUE THREAD

**Issue by yusufkandemir** [vuejs/eslint-plugin-vue#1808]
**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:** 8.9.0
- **eslint-plugin-vue version:** 8.5.0
- **Node version:** 16.14.0
- **Operating System:** Windows 11 21H1 22000.493 - (WSL) Ubuntu 20.04.3 LTS(Kernel: 5.10.93.2-microsoft-standard-WSL2)

**Please show your full configuration:**

```js
/* eslint-env node */
module.exports = {
  root: true,
  extends: [
    "plugin:vue/vue3-recommended",
    "eslint:recommended"
  ],
  env: {
    "vue/setup-compiler-macros": true,
  },
};
```

**What did you do?**	

```vue
<template>
  <div style="color: red;">
    <slot camelCase withValue="foo" :withBind="1" fooBar-BAZ_QuX />
  </div>
</template>
```

**What did you expect to happen?**
I expect warnings like this to appear:
> Attribute 'withVariableBind' must be hyphenated. eslint(vue/attribute-hyphenation)

**What actually happened?**

Nothing happens.

**Repository to reproduce this issue**

https://github.com/yusufkandemir/eslint-plugin-vue-slot-issue

Might be related to https://github.com/vuejs/eslint-plugin-vue/commit/602d80faad9b6004d040209c41fba68e062891d2

`<slot>` is a native HTML element(_related to Web Components_), but the rule should also run on that, just like it used to do before.
